### PR TITLE
Add documentation for the NATS server clustering protocol.

### PR DESCRIPTION
**Pull request by ColinSullivan1**
Documentation for the NATS server protocol (server to server messages), including clustering.

@kozlovic @aricart @lparis 

Resolves https://github.com/nats-io/gnatsd/issues/334.

### CODE REVIEW

File: content/documentation/internals/nats-server-protocol.md [hunk 1]
```diff
@@ -0,0 +1,263 @@
...
+### Syntax
+
+`INFO {["option_name":option_value],...}`
+
+The valid options are as follows:
+
+* `server_id`: The unique identifier of the NATS server
+* `version`: The version of the NATS server
+* `go`: The version of golang the NATS server was built with
+* `host`: The IP address of the NATS server host
```
Comment (kozlovic) [inline content/documentation/internals/nats-server-protocol.md]: In the server's INFO, it is worth noting that `host` and `port` here refer to values specified by the `--cluster` command line parameter or `Options.ClusterHost` and `Options.ClusterPort`.

File: content/documentation/internals/nats-server-protocol.md [hunk 2]
```diff
@@ -0,0 +1,263 @@
...
+Handling of these errors usually has to be done asynchronously.
+
+Protocol error messages which close the connection:
+
+- `-ERR 'Unknown Protocol Operation'`: Unknown protocol error
+- `-ERR 'Authorization Violation'`: Server failed to authenticate to another server with credentials specified in the [CONNECT](#CONNECT) message.
+- `-ERR 'Authorization Timeout'`: Server took too long to authenticate to another server after establishing a connection (default 1 second)
+- `-ERR 'Parser Error'`: Cannot parse the protocol message sent by another server
+- `-ERR 'Stale Connection'`: PING/PONG interval expired.
+- `-ERR 'Slow Consumer'`: The server pending data size for the connection has reached the maximum size (default 10MB).
```
Comment (kozlovic) [inline content/documentation/internals/nats-server-protocol.md]: I think slow consumer is about deadline (time based, default 2 seconds) in sending data to a tcp connection. At least this is what it is for server to clients. Is it different for routes?

**Comment by kozlovic** [comment]
2 small comments, otherwise LGTM

File: content/documentation/internals/nats-server-protocol.md (hunk 1 above)
Comment (ColinSullivan1) [inline content/documentation/internals/nats-server-protocol.md]: Will fix - thanks!

File: content/documentation/internals/nats-server-protocol.md (hunk 2 above)
Comment (ColinSullivan1) [inline content/documentation/internals/nats-server-protocol.md]: Good point - I made an assumption - will look into it.

Comment (ColinSullivan1) [inline content/documentation/internals/nats-server-protocol.md]: I did verify that route connections do detect slow consumers and behave accordingly.

Comment (kozlovic) [inline content/documentation/internals/nats-server-protocol.md]: My comment was not about that `Slow Consumer` not apply to route, but the threshold to consider it a slow consumer is not about maximum size as far as I know, but time it takes to do a socket write. I could not find in the code the reference to a 10MB limit. Could you please point me there?

Comment (wallyqs) [inline content/documentation/internals/nats-server-protocol.md]: Think @ColinSullivan1 refers to examples such as... https://github.com/nats-io/gnatsd/blob/01b635cc8ffbb4054e84cff927f48ea969bd3396/server/configs/test.conf#L37
(though don't see `MaxPending` implemented check around either?)

Comment (ColinSullivan1) [inline content/documentation/internals/nats-server-protocol.md]: I thought you meant applied to routes in general.  It looks to be only the deadline (for either type); I'll remove the note about the default - thanks.

**Comment by kozlovic** [comment]
LGTM

**Comment by ColinSullivan1** [comment]
Any other commenters?  Otherwise, @chriscasper, if this looks good to you we can merge.

**Pull request merged**